src/audio/kpb.c

```diff
@@ -1715,6 +1845,65 @@ static inline bool kpb_is_sample_width_supported(uint32_t sampling_width)
 	return ret;
 }
 
+#if KPB_HIFI4
+static void kpb_copy_24b_in_32b(const struct audio_stream __sparse_cache *source, uint32_t ioffset,
+				struct audio_stream __sparse_cache *sink, uint32_t ooffset,
+				uint32_t n_samples)
+{
+	int ssize = audio_stream_sample_bytes(source); /* src fmt == sink fmt */
+	uint8_t *in = audio_stream_wrap(source, (uint8_t *)source->r_ptr + ioffset * ssize);
+	uint8_t *out = audio_stream_wrap(sink, (uint8_t *)sink->w_ptr + ooffset * ssize);
+
+	const ae_int32x2 *sin = (const ae_int32x2 *)(in);
+	ae_int32x2 *sout = (ae_int32x2 *)(out);
+	ae_int32x2 vs = AE_ZERO32();
+
+	if (!IS_ALIGNED((uintptr_t)sin, 8)) {
+		AE_L32_IP(vs, (const ae_int32 *)(sin), 4);
+		AE_S32_L_IP(vs, (ae_int32 *)(sout), 4);
```

Comment:
you really don't need parentheses around `sin` / `sout` / `in` / `out` - there are multiple such locations in the code